app/src/main/java/com/teamagam/gimelgimel/app/map/esri/EsriGGMapView.java

```diff
@@ -389,6 +403,45 @@ private int getClickedGraphicId(float screenX, float screenY) {
         private boolean isEntityClicked(int graphicId) {
             return graphicId != -1;
         }
+
+        private void handleKmlEntityClicks(float screenX, float screenY) {
+            KmlEntityInfo info = getClickedKmlInfo(screenX, screenY);
+            if (info != null) {
+                notifyKmlEntityClicked(info);
+            }
+        }
+
+        private KmlEntityInfo getClickedKmlInfo(float screenX, float screenY) {
+            Set<String> layerIds = mVectorLayerIdToKmlLayerMap.keySet();
```

Comment:
This is dangerous.
The keyset is backed by the map data, so if its data is chaging so will the set.
while iterating through the set, the collection cannot be modified, an exception will thrown otherwise.
You can resolve this by making a copy of the keyset and work with it. Making a copy can be done with toArray